deeppavlov/models/torch_bert/torch_bert_squad.py

```diff
@@ -150,24 +149,26 @@ def __call__(self, features: List[InputFeatures]) -> Tuple[List[int], List[int],
             predictions: start, end positions, start, end logits positions
 
         """
-        input_ids = [f.input_ids for f in features]
-        input_masks = [f.attention_mask for f in features]
-        input_type_ids = [f.token_type_ids for f in features]
+        _input = {}
+        for elem in ['input_ids', 'attention_mask', 'token_type_ids']:
+            _input[elem] = [getattr(f, elem) for f in features]
 
-        b_input_ids = torch.cat(input_ids, dim=0).to(self.device)
-        b_input_masks = torch.cat(input_masks, dim=0).to(self.device)
-        b_input_type_ids = torch.cat(input_type_ids, dim=0).to(self.device)
+        for elem in ['input_ids', 'attention_mask', 'token_type_ids']:
+            _input[elem] = torch.cat(_input[elem], dim=0).to(self.device)
 
         with torch.no_grad():
             # Forward pass, calculate logit predictions
-            outputs = self.model(input_ids=b_input_ids, attention_mask=b_input_masks, token_type_ids=b_input_type_ids)
+            _model_input = {key:value for (key,value) in _input.items() if key in self.model.forward.__code__.co_varnames}
+            outputs = self.model(**_model_input)
             logits_st, logits_end = outputs[:2]
 
-            bs = b_input_ids.size()[0]
-            seq_len = b_input_ids.size()[-1]
+            bs = _input['input_ids'].size()[0]
+            seq_len = _input['input_ids'].size()[-1]
```

Comment:
You don't have to slice it. The relevant dimension can be accessed by simply passing it's index to the size method. For example, `_input['input_ids'].size(0)` would work the same as `_input['input_ids'].size()[0]`.